storm-client/src/jvm/org/apache/storm/utils/VersionInfo.java

```diff
@@ -172,7 +172,10 @@ private static IVersionInfo getFromClasspath(List<String> classpath, final Strin
                 try {
                     Path parent = p.getParent();
                     List<String> children = new ArrayList<>();
-                    Files.list(parent).forEach(path -> children.add(path.toString()));
+                    Files.list(parent)
+                        //avoid infinite recursion
```

Comment:
infinite recursion can happen if there is a file named `*` under `/<parent-path>/` directory

Reply:
Good finding.